ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -7175,63 +7175,97 @@ void mergeThorGraphs(WorkflowItem & workflow, bool resourceConditionalActions, b
 }
 
 //------------------------------------------------------------------------
-//#define NEW_SCALAR_CODE
-//I think NEW_SCALAR_CODE should be better - but in practice it seems to be worse.....
-
 inline bool isTypeToHoist(ITypeInfo * type)
 {
     return isSingleValuedType(type);// || (type && type->getTypeCode() == type_set);
 }
 
+//Currently we do not create globals for common sub expressions between global expressions
+//I suspect a better solution is to create them much later in the resourcing, but keep the
+//code for the moment for reference.
+static const bool createGlobalForGlobalCse = false;
+
 static HqlTransformerInfo scalarGlobalTransformerInfo("ScalarGlobalTransformer");
 ScalarGlobalTransformer::ScalarGlobalTransformer(HqlCppTranslator & _translator)
 : HoistingHqlTransformer(scalarGlobalTransformerInfo, CTFtraverseallnodes), translator(_translator)
 {
-    okToHoist = true;
-    neverHoist = false;
+    isGlobal = true;
+    isOkToHoist = true;
+}
+
+bool ScalarGlobalTransformer::isCandidate(IHqlExpression * expr)
+{
+    //  Commented line has problems with SELF used in HOLE definition, and explosion in thumphrey7 etc.
```

Comment:
Is the purpose of the commented line to indicate to future coders that the simpler version might seem like it would work (but won't), or to indicate a version that you think should work (but doesn't) that you intend to revisit?

Either way a clearer comment might help.